proposals/0024-opacity-micromaps.md

```diff
@@ -72,6 +72,26 @@ no OMMs present, since it may incur a small penalty on traversal performance
 overall.  See the D3D12 flag definition for more details:
 [`D3D12_RAYTRACING_PIPELINE_FLAG_ALLOW_OPACITY_MICROMAPS`][dxr-flags].
 
+`RayQuery` objects, used with [RayQuery::TraceRayInline()][rq-trace], 
+also need to be aware that OMMs may be in use.  The above `RAYTRACING_PIPELINE_FLAG_ALLOW_OPACITY_MICROMAPS` 
+doesn't apply for inline raytracing however. `RayQuery` objects are 
+independent of raytracing pipelines. For `RayQuery` the template for 
+instantiating the object includes a new optional `RAYQUERY_FLAGS` parameter:
+
+`RayQuery<RAY_FLAGS, RAYQUERY_FLAGS>`
```

Comment:
We need to expand the details of this into the rest of the proposal.  For instance:
- Changes under `Interchange Format Additions`:
  - need a new DXIL op (new version of AllocateRayQuery) to capture the new flags argument.  Decide whether to use the existing op when `RAYQUERY_FLAG_NONE`.  My preference is that we use the old op if we do not require the new op.
  - need new `RayQueryFlag` enum in `DXILConstants.h`
- Update `Diagnostic Changes` with any new relevant diagnostic cases
  - Warn about use of this flag on `SM < 6.9`
  - Do we need to validate that a RayQuery object uses this flag when passing the new `RayFlag::ForceOMM2State` (if constant)?
- Update `Validation Changes` for new op and flag
- Update `Runtime Additions`
  - Given this usage in RayQuery, we may require a new optional feature bit for the FeatureInfo part (since it cannot be determined by DXR state objects in the runtime).
  - After discussion, since it's benign to use the flag on a device that does not support OMM, the feature flag will not be added, and the runtime will not validate usage with feature support.
  - We will need to be clear for the intrinsic op definition that any driver supporting SM 6.9 must support the new DXIL op but may ignore the new `RayQueryFlag` if they do not support OMM.
- Update `Testing` with implications of all these changes.

We can also capture this feedback in an issue and merge the current PR without these updates for now.